common/remote/sockfile.cpp

```diff
@@ -1988,6 +2009,108 @@ extern IFileIO *createRemoteFilteredFile(SocketEndpoint &ep, const char * filena
     return nullptr;
 }
 
+class CRemoteFilteredKeyIO : public CRemoteFilteredFileIOBase
+{
+public:
+    // Really a stream, but life (maybe) easier elsewhere if looks like a file
+    // Sometime should refactor to be based on ISerialStream instead - or maybe IRowStream.
+    CRemoteFilteredKeyIO(SocketEndpoint &ep, const char *filename, IOutputMetaData *actual, IOutputMetaData *projected, const RowFilter &fieldFilters, unsigned __int64 rowLimit)
+        : CRemoteFilteredFileIOBase(ep, filename, actual, projected, fieldFilters)
+    {
+        request.appendf(",\n \"kind\" : \"indexread\"");
```

Comment:
Maybe something to consider for future, but in discussions with John yesterday it occurred to me that really the caller may not want/need to treat index reads differently from diskreads... Can/should dafilesrv automatically process the request correctly depending on the physical file?

Reply:
At some point, we were planning on laying down the meta info. physically on disk next to the parts. In which case dafilesrv could just pick it up.

Until then, it _could_ open the file and deduce it's an index probably...

Reply:
There's a function I added that allows us to spot index files (used by Roxie in standalone mode when no dali present) - you could use that if you wanted